pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -57,975 +56,801 @@
-                )
-            # this automatic switch is to ease transition between accelerator environments
+        if self._precision_flag == 16 and self._accelerator_flag == "cpu":
             rank_zero_warn(
                 "You passed `Trainer(accelerator='cpu', precision=16)` but native AMP is not supported on CPU."
                 " Using `precision='bf16'` instead."
             )
-            self.precision = "bf16"
-
-        if self.precision in (16, "bf16"):
-            if self.precision == "bf16" and self.amp_type != AMPType.NATIVE:
-                raise MisconfigurationException(
-                    f"You passed `Trainer(amp_type={self.amp_type.value!r}, precision='bf16')` but it's not supported."
-                    " Try using `amp_type='native'` instead."
-                )
+            self._precision_flag = "bf16"
 
+        if self._precision_flag in (16, "bf16"):
             rank_zero_info(
-                f"Using 16bit {self.amp_type.value} Automatic Mixed Precision (AMP)"
-                if self.precision == 16
+                f"Using 16bit {self._amp_type_flag.value} Automatic Mixed Precision (AMP)"
+                if self._precision_flag == 16
                 else "Using bfloat16 Automatic Mixed Precision (AMP)"
             )
 
-            if self.amp_type == AMPType.NATIVE:
-                device = "cpu" if self.use_cpu else "cuda"
+            if self._amp_type_flag == AMPType.NATIVE:
+                device = "cpu" if self._accelerator_flag == "cpu" else "cuda"
 
-                if self._is_sharded_training_type:
-                    return ShardedNativeMixedPrecisionPlugin(self.precision, device)
-                if self._is_fully_sharded_training_type:
-                    return FullyShardedNativeMixedPrecisionPlugin(self.precision, device)
-                return NativeMixedPrecisionPlugin(self.precision, device)
+                if isinstance(self.strategy, (DDPShardedStrategy, DDPSpawnShardedStrategy)):
+                    return ShardedNativeMixedPrecisionPlugin(self._precision_flag, device)
+                if isinstance(self.strategy, DDPFullyShardedStrategy):
+                    return FullyShardedNativeMixedPrecisionPlugin(self._precision_flag, device)
+                return NativeMixedPrecisionPlugin(self._precision_flag, device)
 
-            if self.amp_type == AMPType.APEX:
-                if self._is_sharded_training_type or self._is_fully_sharded_training_type:
-                    raise MisconfigurationException(
-                        "Sharded plugins are not supported with apex, please switch to `amp_backend='native'`."
-                    )
-                self.amp_level = self.amp_level or "O2"
-                return ApexMixedPrecisionPlugin(self.amp_level)
+            if self._amp_type_flag == AMPType.APEX:
+                self._amp_level_flag = self._amp_level_flag or "O2"
+                return ApexMixedPrecisionPlugin(self._amp_level_flag)
```

Comment:
```suggestion
            if self._amp_type_flag == AMPType.APEX:
                return ApexMixedPrecisionPlugin(self._amp_level_flag)
```

This is already the default value in the `ApexMixedPrecisionPlugin` plugin.

Reply:
updated